backend/src/rss.rs

```diff
@@ -0,0 +1,243 @@
+        .encoding("UTF-8".into())
+        .build();
+
+    let mut rss = XMLElement::new("rss");
+
+    // Add rss attributes
+    let attributes = [
+        ("version", "2.0"),
+        ("xmlns:dc", "http://purl.org/dc/elements/1.1/"),
+        ("xmlns:content", "http://purl.org/rss/1.0/modules/content/"),
+        ("xmlns:atom", "http://www.w3.org/2005/Atom"),
+        ("xmlns:media", "http://search.yahoo.com/mrss/"),
+        ("xmlns:itunes", "http://www.itunes.com/dtds/podcast-1.0.dtd"),
+    ];
+    for (name, value) in attributes {
+        rss.add_attribute(name, value);
+    }
+
+    let mut channel = XMLElement::new("channel");
+
+    // Add channel tags
+    let channel_data = [
+        ("title", vec![("", series_title.clone())]),
+        ("link", vec![("", series_link)]),
+        ("description", vec![("", series_description.unwrap_or_default())]),
+        ("language", vec![("", "und".to_string())]),
+        ("itunes:explicit",vec![("", "true".to_string())]),
+        ("itunes:image", vec![("href", (&cover_image.to_string_lossy()).to_string())]),
+        ("itunes:category", vec![("text", "Education".to_string())]),
+        ("atom:link", vec![
+            ("href", rss_link.clone()),
+            ("rel", "self".to_string()),
+            ("type", "application/rss+xml".to_string()),
+        ]),
+    ];
+    add_elements_to_xml(&channel_data, &mut channel);
+
+    // Add video items
+    let video_items = match generate_video_items(
+        &db, &series_oc_id, &series_title, &rss_link, &tobira_url).await {
+        Ok(items) => items,
+        Err(_) => return Err(bad_request("empty series")),
+    };
+
+    for item in video_items {
+        // I have no idea how to correctly handle items that are resulting in an error.
+        // Pretty sure however that this here is a bad solution and would result in
+        // potentially broken feeds.
+        let _ = channel.add_child(item);
+    }
+
+    rss.add_child(channel).unwrap();
+    xml.set_root_element(rss);
+    let mut writer: Vec<u8> = Vec::new();
+    xml.generate(&mut writer).unwrap();
+
+    let xml_string = String::from_utf8(writer).unwrap();
+    
+    Ok(xml_string)
+}
+
+/// Generates the single video items of a series in Tobira for inclusion in an RSS feed.
+async fn generate_video_items(
+    db: &Client,
+    series_oc_id: &str,
+    series_title: &str,
+    rss_link: &str,
+    tobira_url: &HttpHost,
+) -> Result<Vec<XMLElement>, Error> {
+    let selection = Event::select();
+    let query = format!("select {selection} from events where part_of = $1");
+    let rows = db.query_raw(&query, dbargs![&series_oc_id]).await?;
+
+    let mut video_items: Vec<XMLElement> = Vec::new();
+
+    rows.try_for_each(|row| {
+        let event = Event::from_row_start(&row);
+        
+        let mut buf = [0; 11];
+        let tobira_event_id = event.id.to_base64(&mut buf);
+        let event_link = format!("{}/!v/{}", tobira_url, tobira_event_id);
```

Comment:
Only noticing this now: this can be written a bit ncier:

```suggestion
        let event_link = format!("{tobira_url}/!v/{tobira_event_id}");
```

Maybe you can use that trick elsewhere in this file as well.